modules/caddyhttp/server.go

```diff
@@ -209,6 +209,13 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// restore original request before invoking error handler chain (issue #3717)
+	origReq := r.Context().Value(OriginalRequestCtxKey).(http.Request)
+	r.Method = origReq.Method
+	r.RemoteAddr = origReq.RemoteAddr
+	r.RequestURI = origReq.RequestURI
+	cloneURL(origReq.URL, r.URL)
+
```

Comment:
What about the request headers if someone uses `request_header`? :thinking:

Reply:
Good point... we don't currently copy those into the OriginalReq as it is, and I've never heard of a request (pun intended) for this. Because it's a map of variable size I am cautious of its performance implications, esp. since some requests can be large (with cookies and what-not).

Maybe we just wait and see if it's a problem for anyone?

Reply:
Yeah that's fair. Should we just add a TODO comment or whatever here to note that?

Reply:
Sure, done